fauxfactory/__init__.py

```diff
@@ -520,25 +520,50 @@ def gen_negative_integer():
     return gen_integer(max_value=max_value)
 
 
-def gen_ipaddr(ip3=False, ipv6=False):
+def gen_ipaddr(ip3=False, ipv6=False, prefix=None):
     """Generates a random IP address.
 
+    You can also specify an IP address prefix if you are interested in
+    local network address generation, etc.
+
     :param bool ip3: Whether to generate a 3 or 4 group IP.
     :param bool ipv6: Whether to generate IPv6 or IPv4
+    :param list prefix: A prefix to be used for IPv4 (like [10, 0, 1])
     :returns: An IP address.
     :rtype: str
+    :raises: ``ValueError`` if ``prefix`` is too long either to fit into
+        the IP address or disabling randomness at all.
```

Comment:
@mfalesni The code will not be orthogonal if you allow an empty prefix to be specified. A user can specify an emtpy prefix either by passing in `None` or by passing in an empty iterable such as `[]` or `()`. Non-orthogonal code is harder to understand and more complex.

Now, if you do want to allow empty prefixes, that's fine. It makes a lot of sense, actually. But in that case, disallow `None` and instead write this:

    def gen_ipaddr(ip3=False, ipv6=False, prefix=()):

If you do that, then there is only one way to specify an empty prefix: by passing in an empty iterable.